DXMainClient/Domain/Multiplayer/CnCNet/CnCNetTunnel.cs

```diff
@@ -85,6 +85,29 @@ public static CnCNetTunnel Parse(string str)
         public double Distance { get; private set; }
         public int PingInMs { get; set; } = -1;
 
+        public int Rating
+        {
+            get
+            {
+                if (Clients + 24 >= MaxClients)
+                    return int.MaxValue;
+
+                if (Official || Recommended)
+                {
+                    if (PingInMs <= -1)
+                        return int.MaxValue - 200000 + Clients;
+
+                    int ping = PingInMs < 1 ? 1000 : PingInMs;
+
+                    return (int)(Math.Ceiling((double)ping / 70) * 100000) + Clients;
```

Comment:
Would be good to have a comment explaining this stuff (even better if it was a docstring for constants to be used instead of magic numbers and the property itself). The first comment about quadratic/exponential function also applies here. Also magic numbers again.

Reply:
Named constants with comments have been defined based on this recommendation.